pandas/io/formats/format.py

```diff
@@ -1767,16 +1767,13 @@ def _format_datetime64_dateonly(
     nat_rep: str = "NaT",
     date_format: str | None = None,
 ) -> str:
-    if x is NaT:
+    if x is NaT or isinstance(x, NaTType):
```

Comment:
why is NaTType leaking here? (is this is a typing issue? e.g. maybe cast?)

Reply:
I think it is better to just do `if isinstance(x, NaTType)` here - the typing makes it clear.  Will be in next commit.

Reply:
checking for both NaT and NaTType is redundant.  the existing check should be marginally more performant than an isintance check.

Reply:
The issue here is that `x` is of type `NaTType | Timestamp`, so if you just do `x is NaT` then `mypy` can't conclude that `x` is `Timestamp` when the condition is `False`, but if you do `isinstance(x, NaTType)`, it can make that conclusion when the condition is `False`.

Reply:
im aware.  i dont think that justifies making the check redundant.

Reply:
That's why I just changed it to `if isinstance(x, NaTType)` in https://github.com/pandas-dev/pandas/pull/46171/commits/e79d2536094c0b5c64b50910343670d46965f1df

Reply:
as commented elsewhere, -1 on this change.